clients/settings/permissions.go

```
@@ -0,0 +1,341 @@
+	if err != nil {
+		return ErrorPermissionCreate{err}
+	}
+
+	return err
+}
+
+// Update updates the permissions of the accessor for the settings object identified by the provided objectID.
+// The function sends a PUT request to the server using the given context, objectID, and permissions.
+// It returns an error indicating the success or failure of its execution.
+//
+// This is not an upsert operation. If there are no permissions set for all-users accessor yet,
+// this call will fail and Create needs to be used. This needs to be handled by the caller.
+//
+// # If the HTTP request to the server fails, the method returns an error
+//
+// If you wish to receive logs from this method supply a logger inside the context using logr.NewContext.
+//
+// Parameters:
+//   - ctx: Context for controlling the HTTP operation's lifecycle.
+//   - objectID: The id of the settings object for which the permissions are to be created.
+//   - permissions: contains the accessor and the permissions which are to be granted to accessor.
+//
+// Returns:
+//   - error: An error if the HTTP call fails or another error happened.
+func (d *Client) Update(ctx context.Context, objectID string, permission PermissionObject) error {
+	if objectID == "" {
+		return ErrorMissingObjectId
+	}
+
+	payload, err := json.Marshal(permission)
+	if err != nil {
+		return ErrorMarshalling{err}
+	}
+
+	path, err := constructAPIPath(objectID, permission.Accessor)
+	if err != nil {
+		return err
+	}
+
+	httpResp, err := d.client.PUT(
+		ctx,
+		path,
+		bytes.NewReader(payload),
+		rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
+
+	if err != nil {
+		return ErrorPermissionUpdate{err}
+	}
+
+	_, err = api.NewResponseFromHTTPResponse(httpResp)
+	if err != nil {
+		return ErrorPermissionUpdate{err}
+	}
+
+	return err
+}
+
+// Delete deletes the permissions of the provided accessor for the settings object identified by the provided objectID.
```

Maybe also mention that 404 errors are ignored

---

Is there a reason we ignore 404s?

---

All this will be moved away from the core lib.